packages/flame/lib/src/game/base_game.dart

```diff
@@ -87,13 +87,34 @@ class BaseGame extends Game with FPSCounter {
     _combinedProjector = Projector.compose([camera, viewport]);
   }
 
+  /// This method setps up the OrderedSet instance used by this game, before
+  /// any lifecycle methods happen.
+  ///
+  /// You can return a specific sub-class of OrderedSet, like
+  /// [QueryableOrderedSet] for example, that we use for Collidables.
+  OrderedSet<Component> createOrderedSet() {
```

Comment:
I always wanted to have this. For very complex games, this allow more easily users to override and return very customized stuff (we do that on Tales for example but we replace the components variable that now is final)